packages/medusa/src/commands/develop.js

```diff
@@ -23,6 +23,109 @@ export default async function ({ port, directory }) {
   args.shift()
   args.shift()
 
+  /**
+   * Re-constructing the path to Medusa CLI to execute the
+   * start command.
+   */
+  const cliPath = path.resolve(
+    require.resolve("@medusajs/medusa-cli"),
+    "..",
+    "..",
+    "cli.js"
+  )
+
+  const devServer = {
+    childProcess: null,
+    watcher: null,
+
+    /**
+     * Start the development server by forking a new process.
+     *
+     * We do not kill the parent process when child process dies. This is
+     * because sometimes the dev server can die because of programming
+     * or logical errors and we can still watch the file system and
+     * restart the dev server instead of asking the user to re-run
+     * the command.
+     */
+    start() {
+      this.childProcess = fork(cliPath, ["start", ...args], {
+        cwd: directory,
+        env: {
+          ...process.env,
+          NODE_ENV: "development",
+        },
+        execArgv: argv,
+      })
+      this.childProcess.on("error", (error) => {
+        Logger.error("Dev server died with error", error)
```

Comment:
Suggestion: Dev server cannot start, error

Reply:
Can we instead say "Dev server exited with error"

The reason is "Dev server was running, but it crash or exited during the watch mode". WDYT?

Reply:
Sounds good to me 👌